homeassistant/components/touchline/config_flow.py

```diff
@@ -0,0 +1,104 @@
+"""Config flow for Roth Touchline floor heating controller."""
+from __future__ import annotations
+
+import logging
+import re
+from typing import Any
+
+from pytouchline import PyTouchline
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_HOST
+from homeassistant.data_entry_flow import FlowResult
+import homeassistant.helpers.config_validation as cv
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_HOST): str,
+    }
+)
+
+RESULT_SUCCESS = "success"
+RESULT_CANNOT_CONNECT = "cannot_connect"
+
+
+def _try_connect_and_fetch_basic_info(host):
+    """Attempt to connect and, if successful, fetch number of devices."""
+    py_touchline = PyTouchline()
+    result = {"type": None, "data": {}}
+    number_of_devices = None
+    try:
+        number_of_devices = int(py_touchline.get_number_of_devices(host))
+        if number_of_devices:
+            result["type"] = RESULT_SUCCESS
+            result["data"] = None
```

Comment:
Why are we setting to `None` here? You already initialize it to an empty dict above.

Reply:
Agree. I will remove it!

Reply:
Removed.